homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,254 @@
+
+    return DaikinHVAC(device, name, customize)
+
+
+class DaikinHVAC(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, device, name, customize):
+        import pydaikin.appliance as appliance
+
+        """Initialize the HVAC."""
+        self._name = name
+        self._device = device
+        self._customize = customize
+
+        self._current_temperature = self.settings('htemp', True)
+        self._operation_list = list(DAIKIN_OPERATION_LIST.values())
+        self._fan_list = appliance.daikin_values('f_rate')
+        self._swing_list = appliance.daikin_values('f_dir')
+
+        self._target_temperature = None
+        self._current_humidity = None
+        self._target_humidity = None
+        self._current_fan_mode = None
+        self._current_swing_mode = None
+
+    @property
+    def unique_id(self):
+        """Return the ID of this AC."""
+        return "{}.{}".format(self.__class__, self._device.ip)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def should_poll(self):
+        """Polling needed for a demo climate device."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        self._current_temperature = self.settings('htemp', True)
+
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
+
+        return self._target_temperature
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        if temperature is None:
+            return
+
+        self._device.set({"temp": temperature})
+        self._target_temperature = temperature
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        state = self._device.represent('mode')[1]
+
+        current_operation = DAIKIN_STATE_TO_HA.get(state)
+        if current_operation in [STATE_ECO, STATE_PERFORMANCE]:
+            current_operation = STATE_AUTO
+
+        return current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    def set_operation_mode(self, operation_mode):
+        """Set HVAC mode (auto, auxHeatOnly, cool, heat, off)."""
+
+        current_operation = HA_STATE_TO_DAIKIN.get(operation_mode)
+        self._device.set({"mode": current_operation})
```

Comment:
This will call `_device.set()` with `{"mode": None}` if someone specifies a bogus mode in the service call. Not sure what the library does then, but perhaps you want to catch that

Reply:
It's hard that thing to happend but I've just treated it now :)